docs/spark/how-to-guides/dotnet-interactive-udf-issue.md

```diff
@@ -0,0 +1,65 @@
+Now let's see how that affects our UDF definition in the notebook:
+
+![UDF serialization error](./media/dotnet-interactive/udf-serialization-error.png)
+
+As can be seen in the case of `udf2_fails`, we see the error message which says Type `Submission#7` is not marked as serializable, this is because how .NET Interactive works is it wraps every object defined in a cell with its `Submission#` class which is generated on the fly and hence is not marked as `Serializable`, hence the error.
+For this reason, it is required that a UDF referencing a custom object in it, is defined in the same cell as that object.
+
+## Calling a UDF on a DataFrame
+
+When calling a previously defined UDF on a `DataFrame` it is important to make sure the UDF is defined in a previously submitted cell, before calling it.
+For example, let's use the UDF shown in the previous example and call it on a DataFrame:
+
+![working UDF call](./media/dotnet-interactive/udf_works.png)
+
+Now let's see what happens if we call the UDF in the same cell where it is defined.
+
+![failing UDF call](./media/dotnet-interactive/udf_fails.png)
+
+The above highlighted error is because the UDF assemblies need to first be compiled and shipped to the workers before it can be called on a DataFrame.
+
+## Why Broadcast Variables don't work with .NET Interactive
+
+The above explained behavior is the reason why Broadcast Variables don't work with .NET Interactive. It is a good idea to go through [this guide on Broadcast Variables](broadcast-guide.md) to get a deeper understanding of what Broadcast variables are and how to use them. The reason Broadcast Variables don't work with interactive scenarios is because of .NET interactive's design of appending each object defined in a cell with it's cell submission class, which since is not marked serialzizable, fails with the same exception as shown previously.
+Let's dive a little deeper with the following example:
+
+![Broadcast Variables fail](./media/dotnet-interactive/broadcast-fails.png)
+
+As recommended in the previous sections, we define both the UDF and the object it is referencing (Broadcast variable in this case) in the same cell, but we still see the `SerializationException` error complaining about `Microsoft.Spark.Sql.Session` not being marked as serializable. This is because when the compiler tries to serialize the Broadcast variable object `bv`, it finds its name to be appended with [`SparkSession`](https://github.com/dotnet/spark/blob/master/src/csharp/Microsoft.Spark/Sql/SparkSession.cs#L20) object `spark` which it requires to be marked as serializable. This can be demonstrated with more ease by taking a peek at the decompiled assembly of this cell submission:
```

Comment:
```suggestion
As recommended in the previous sections, we define both the UDF and the object it is referencing (broadcast variable in this case) in the same cell, but we still see the `SerializationException` error complaining about `Microsoft.Spark.Sql.Session` not being marked as serializable. This is because when the compiler tries to serialize the broadcast variable object `bv`, it finds its name to be appended with [`SparkSession`](https://github.com/dotnet/spark/blob/master/src/csharp/Microsoft.Spark/Sql/SparkSession.cs#L20) object `spark` which it requires to be marked as serializable. This can be demonstrated with more ease by taking a peek at the decompiled assembly of this cell submission:
```